core/src/main/java/cucumber/runner/TimeService.java

```diff
@@ -1,28 +1,70 @@
 package cucumber.runner;
 
 public interface TimeService {
-    long time();
+    // The reason to have three method, which actually is doing the same thing,
+    // is to enable stubbing of the time service in tests. With the distinction
+    // between startTime() and finishTime() it is easy for the stub to make sure
+    // the the duration between the start time and finish time of the test case,
+    // is the sum of the durations of the test steps.
```

Comment:
I'd like to think of a better way. Test concerns shouldn't leak into interface definitions.